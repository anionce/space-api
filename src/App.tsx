import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { MainLayout } from './layout';
import { ItemDetails } from './containers/ItemDetails';
import { AddRocket } from './containers/AddRocket/AddRocket';
import './services/i18n';
import './App.scss';
import { NotFound } from './containers/NotFound';
import { Home } from './containers/Home';
import { ListPage } from './containers/ListPage/ListPage';

export const App = () => {
	return (
		<Router>
			<Routes>
				<Route element={<MainLayout />}>
					<Route index element={<Home />} />
					<Route path='capsules' element={<ListPage />} />
					<Route path='capsules/:id' element={<ItemDetails />} />
					<Route path='rockets' element={<ListPage />} />
					<Route path='rockets/:id' element={<ItemDetails />} />
					<Route path='rockets/:id/update' element={<AddRocket />} />
					<Route path='rocket/add' element={<AddRocket />} />
					<Route path='*' element={<NotFound />} />
				</Route>
			</Routes>
		</Router>
	);
};
