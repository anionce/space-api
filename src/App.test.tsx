import { render } from '@testing-library/react';
import React from 'react';
import { App } from './App';

test('app renders', () => {
	const { container } = render(<App />);

	expect(container).toMatchSnapshot();
});
