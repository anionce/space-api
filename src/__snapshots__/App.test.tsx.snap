// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders 1`] = `
<div>
  <div
    class="header"
  >
    <a
      href="/"
    >
      <img
        alt="header-logo"
        class="header-logo"
        src="spaceLogo.svg"
      />
    </a>
  </div>
  <nav>
    <a
      class=""
      href="/capsules"
    >
      Capsules
    </a>
    <a
      class=""
      href="/rockets"
    >
      Rockets
    </a>
    <a
      class=""
      href="/rocket/add"
    >
      Add a rocket
    </a>
  </nav>
  <div
    class="home-container"
  >
    <h1>
      Welcome to the space API
    </h1>
    <h2>
      Ready?
    </h2>
  </div>
  <div
    class="footer"
  >
    <img
      alt="footer-logo"
      src="footer.svg"
    />
  </div>
</div>
`;
